src/components/createsecret.test.ts

```diff
@@ -0,0 +1,183 @@
+  });
+});
+test('create secret', async () => {
+  await flushPromises();
+
+  const secretName = wrapper.find<HTMLInputElement>(
+    '[data-test="secretNameInput"]'
+  );
+  const secretKey = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyNameInput-0"]'
+  );
+  const secretValue = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyValueInput-0"]'
+  );
+
+  secretName.setValue('newSec01');
+  secretKey.setValue('new sec key 01');
+  secretValue.setValue('sdsdsdsds sdsdsdsd H m,mjhn9');
+
+  wrapper.find('form').trigger('submit.prevent');
+
+  await flushPromises();
+
+  expect(wrapper.router.push).toHaveBeenCalledTimes(1);
+  expect(wrapper.router.push).toHaveBeenCalledWith({
+    path: '/org/org01/projects/proj01.proj/settings',
+  });
+});
+
+test('invalid entries', async () => {
+  await flushPromises();
+
+  const secretName = wrapper.find<HTMLInputElement>(
+    '[data-test="secretNameInput"]'
+  );
+  const secretKey = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyNameInput-0"]'
+  );
+  const secretValue = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyValueInput-0"]'
+  );
+
+  secretName.setValue('');
+  secretKey.setValue('');
+  secretValue.setValue('');
+  expect(wrapper.router.push).not.toHaveBeenCalled();
+  const createButton = wrapper.find('[data-test="secretCreateButton"]');
+  console.log(wrapper.html());
+  expect(createButton.attributes('disabled')).toBeDefined();
+});
+
+test('error response from the server is correctly displayed', async () => {
+  await flushPromises();
+
+  server.use(
+    http.post('/api/v1alpha/projects/org%2Forg01%2Fproj01/secrets', () => {
+      return new HttpResponse(JSON.stringify({ message: 'Server Error' }), {
+        status: 500,
+      });
+    })
+  );
+
+  const secretName = wrapper.find<HTMLInputElement>(
+    '[data-test="secretNameInput"]'
+  );
+  const secretKey = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyNameInput-0"]'
+  );
+  const secretValue = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyValueInput-0"]'
+  );
+
+  secretName.setValue('newSec01');
+  secretKey.setValue('new sec key 01');
+  secretValue.setValue('sdsdsdsds sdsdsdsd H m,mjhn9');
+
+  wrapper.find('form').trigger('submit.prevent');
+
+  await flushPromises();
+
+  expect(wrapper.router.push).not.toHaveBeenCalled();
+
+  const serverError = wrapper.find('[data-test="serverError"]');
+  expect(serverError.text()).toContain('api error');
+});
+
+test('reset component on prop change', async () => {
+  await flushPromises();
+  const secretName = wrapper.find<HTMLInputElement>(
+    '[data-test="secretNameInput"]'
+  );
+  const secretKey = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyNameInput-0"]'
+  );
+  const secretValue = wrapper.find<HTMLInputElement>(
+    '[data-test="secretKeyValueInput-0"]'
+  );
+  secretKey.setValue('new sec key 01');
+  secretValue.setValue('sdsdsdsds sdsdsdsd H m,mjhn9');
+
+  wrapper.setProps({
```

Comment:
you should also check that when setting props to the same current value the reset should not happen.